### PR TITLE
cluster-api-provider-azure: add workflow-based e2e-upgrade test

### PR DESCRIPTION
**Pull request by vrutkovs**
Create necessary steps to install cluster from `stable-initial:installer` on Azure and upgrade it using workflow approach.

### CODE REVIEW

**Comment by vrutkovs** [comment]
SDN crashlooping on upgrade
/retest

**Review by michaelgugino — APPROVED** [review]
/lgtm

**Comment by vrutkovs** [comment]
Workers didn't come up on initial setup

/retest

**Comment by vrutkovs** [comment]
Workers didn't come up on initial setup

/retest

File: ci-operator/step-registry/openshift/upgrade/azure/loki/openshift-upgrade-azure-loki-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    post:
+    - ref: gather-loki
+    - chain: ipi-azure-post
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: "release:initial"
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: "release:latest"
+    env:
+      TEST_COMMAND: "run-upgrade"
+      TEST_SUITE: "all"
+      TEST_OPTIONS: "abort-at=100"
```
Comment (tnozicka) [inline ci-operator/step-registry/openshift/upgrade/azure/loki/openshift-upgrade-azure-loki-workflow.yaml:19]: https://github.com/openshift/release/pull/12544/files

File: ci-operator/step-registry/openshift/upgrade/azure/loki/openshift-upgrade-azure-loki-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    - ref: ipi-install-monitoringpvc
+    - ref: ipi-install-loki
+    - chain: ipi-azure-pre-stableinitial
+    test:
+    - ref: openshift-e2e-test
+    post:
+    - ref: gather-loki
+    - chain: ipi-azure-post
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: "release:initial"
```
Comment (tnozicka) [inline ci-operator/step-registry/openshift/upgrade/azure/loki/openshift-upgrade-azure-loki-workflow.yaml:14]: as latest is older then this PR should it be from latest to initial?

Comment (vrutkovs) [reply ci-operator/step-registry/openshift/upgrade/azure/loki/openshift-upgrade-azure-loki-workflow.yaml:14]: Names are a bit confusing, but `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` is used in `ipi-install-install` (so its initial) and `OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE` used in `openshift-e2e-test` doing `openshift-tests run-upgrade --to-image=${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}`

**Pull request merged**